### ISSUE TITLE
nyan on ubuntu xenial causes lag  on vertical cursor movement

### ISSUE THREAD

**Issue by pcn** [TeMPOraL/nyan-mode#27]
```
$ emacs  --version
GNU Emacs 24.5.1
Copyright (C) 2015 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
```

Enabling nyan on my laptop causes a pretty long delay when scrolling vertically by holding down C-n.  What can I do to provide you with relevant info?

**Comment by TeMPOraL** [comment]
Could you also provide your laptop specs?

**Comment by pcn** [comment]
It's a pretty performant laptop in general.  Dell 9550, 4k screen, quad-core, 16gb RAM, 512gb SSD.  I'm not sure at the moment if it's using accelerated nvidia drivers or not.

**Comment by TeMPOraL** [comment]
Thanks! One more question - was it a change of behaviour since last update, or have you just started using Nyan Mode?

Anyway, I'll look into that soon, and in general into speeding up Nyan Mode.

**Comment by pcn** [comment]
I've had this for about a month, and it's behaved this way consistently.  In the past I've used nyan-mode on a macbook, so I don't have a good basis for comparison.

**Comment by TeMPOraL** [comment]
Ok, thanks for the info! :).

**Comment by TeMPOraL** [comment]
I have some ideas for improvements and will be testing them shortly (I do also need to learn how to profile Emacs...). In the meantime, have you tried switching from default linum-mode to [nlinum-mode](http://elpa.gnu.org/packages/nlinum.html)? It works more-less the same, but gives significant performance improvements when scrolling buffers.

**Comment by pcn** [comment]
I haven't, but I'll look at that.

**Comment by seanirby** [comment]
Thought I would chime in since I was also having issues with intermittent lagging while scrolling.  Turning off 'nyan-mode' mode resulted in a minor improvement but turning off powerline fixed the lag completely.

My Emacs version is 25.2.50.1 and my laptop is a 2013 Macbook Pro Retina(2.4Ghz CPU and 8GB of RAM)

Thanks for the package btw :)

**Comment by pcn** [comment]
I've investigated this more in my setup, and currently on ubuntu zesty with emacs25.  I discovered `M-x profiler-start`.  It turns out that projectile was doing a tremendous amount of extra work on every line move.  I don't know if this was my original problem, but at this point things are much better for me.

I mostly followed this issue for ideas: https://github.com/syl20bnr/spacemacs/issues/4207 and in the end it was enabling caching that makes things better for me.
